### ISSUE TITLE
💬 RFC: Add icons to the Status component

### ISSUE THREAD

**Issue by ShiranHi** [backstage/backstage#23421]
### 🔖 Need

Currently, we rely on colored dots in addition to the status name. However, below is a screenshot captured with the color blind deuteranomaly setting, highlighting the issue with using these colored dots. It appears that finding a better solution may be necessary.

![Light mode](https://github.com/backstage/backstage/assets/77277517/fcf37a16-23c0-424a-a73d-18286c17a521)

<img width="503" alt="image" src="https://github.com/backstage/backstage/assets/77277517/595b38a5-5626-42dc-bbe3-413c8605d259">

### 🎉 Proposal

I propose integrating icons from the MUI library that symbolize the status. I used the same colors and I think this is something to be changes as part of the theme. 
<img width="526" alt="Statuses" src="https://github.com/backstage/backstage/assets/77277517/b4b5bca9-c379-45b4-a468-aad7b9e089d6">

### 〽️ Alternatives

I explored also the outlined icons and other colors
<img width="751" alt="Statuses other" src="https://github.com/backstage/backstage/assets/77277517/cb3279cd-f3d8-4590-8b5b-18ca07968d25">

### 👀 Have you spent some time to check if this RFC has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

**Comment by emmaindal** [comment]
Hello @ShiranHi! Thank you so much for opening this issue! I love the level of detail and the designs you have provided, as well as the alternatives you have presented. It really helps!

This feels like such an improvement to me, and I'll run these ideas by our designers and see what they think, or if they have any feedback to provide.

You can expect to hear back from us in the next week or so. Thank you!

**Comment by UsainBloot** [comment]
This change could also be important for users who are colour blind. The current implementation of coloured circles would mean that they can not distinguish between the different states easily. This proposed change helps with that 👍

**Comment by awanlin** [comment]
Really like the proposed new values with the icons! 🚀

**Comment by freben** [comment]
We'd love to see this contributed!

**Comment by awanlin** [comment]
Agree with all the above people, @ShiranHi this seems like a great starting point for a contribution. You have a few open issues but most of them I think still need feedback where is this one if pretty solid and can be actioned. When can you get started on this one?

**Comment by ShiranHi** [comment]
@awanlin sounds good. I'm a designer, so I'll need a hand from someone who can help us implementing this. Do you need something from me? I can share the Figma file

**Comment by awanlin** [comment]
Oh, just thought of something else, where are the icons for this? They should be from the Material UI set but we should list out which ones to use, that would be helpful for who every picks this up.

**Comment by awanlin** [comment]
Perfect, thanks @yuvalgalanti

**Comment by npiyush97** [comment]
@awanlin how these look ? https://carbondesignsystem.com/patterns/status-indicator-pattern/

**Comment by npiyush97** [comment]
> @npiyush97, would much rather pull from Material UI as that's what we are using already and they said that's where they for them for the proposed design so it's all good 👍

how about assets folder in which add required svg and import them 
image added here just testing svg with ::before backgroundimage
![Screenshot from 2024-04-21 22-24-17](https://github.com/backstage/backstage/assets/43876655/e591f4c9-7544-4398-8bf8-db171c6fb3b2)
```
import success from './assets/success.svg'

    ok: {
      '&::before': {
        content: '""',
        backgroundColor: theme.palette.status.ok,
        backgroundImage: `url("${success}")`
      },
    },

```
custom SVG would be better than library ?
